File: App/app/src/main/java/com/prm392g2/prmapp/api/dtos/decks/DeckListArgumentsDTO.java
package com.prm392g2.prmapp.api.dtos.decks;

public class DeckListArgumentsDTO
{
    public String search;
    public Integer minCardCount;
    public Integer maxCardCount;
    public DeckListMetric sortingMetric;
    public boolean sortingAscending;
    public int pageIndex;
    public int pageSize;
}

File: Server/PRMServer.Application/DTOs/Decks/DeckListArgumentsDTO.cs
namespace PRMServer.Application.DTOs.Decks
{
    public record DeckListArgumentsDTO
    {
        public string? Search { get; set; } = null;
        public int? MinCardCount { get; set; } = null;
        public int? MaxCardCount { get; set; } = null;
        public DeckListMetric SortingMetric { get; set; } = DeckListMetric.Name;
        public bool SortingAscending { get; set; } = true;
        
        //public int PageIndex { get; set; } = 0;
        //public int PageSize { get; set; } = 10;
    }

    public enum DeckListMetric
    { 
        Name,
        View,
        Download,
    }
}
